great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -961,6 +1019,34 @@ def __init__(
             usage_statistics_handler=usage_statistics_handler,
         )
 
+    def save_profiler(
```

Comment:
@Shinnnyshinshin I like this approach (creates a new configuration after reconciling it).  I do, however, feel a little uneasy about not using the top-lever reconciliation method calls.  If that can be done, that would be much easier to maintain long term.  This is important, hence I would be happy to discuss (because it requires a tiny bit more work).  Thanks.

Reply:
So this was also negated by a modification to the `add_rule()` method. Basically the`add_rule()` method does all th conversion back and forth between `dict` and `Rule` and ensures that both the profiler's `.rule` field and `._profiler_config.rules` are updated. That means the `save_profiler()` method in `DataContext` turns into 

```
key: Union[
            GeCloudIdentifier, ConfigurationIdentifier
        ] = self.profiler_store.determine_key(name=name, ge_cloud_id=ge_cloud_id)
        self.profiler_store.set(key=key, value=profiler.config)
```
These two lines